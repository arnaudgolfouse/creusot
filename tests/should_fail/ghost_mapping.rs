extern crate creusot_contracts;
use creusot_contracts::{logic::Mapping, *};

#[logic(open(self), prophetic)]
pub fn f(x: &mut i32) -> Mapping<(), i32> {
    pearlite! { |_| ^x }
}

#[logic(open(self))]
pub fn g(x: &mut i32) -> Mapping<(), i32> {
    pearlite! { |_| ^x }
}

#[logic(open(self), prophetic)]
pub fn h(y: &mut i32) -> bool {
    pearlite! { forall<_x:Int> ^y == 1i32 }
}

#[logic(open(self))]
pub fn i(y: &mut i32) -> bool {
    pearlite! { forall<_x:Int> ^y == 1i32 }
}
