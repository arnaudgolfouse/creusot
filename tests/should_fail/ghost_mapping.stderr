error: called prophetic logic function `<&mut T as creusot_contracts::logic::ops::Fin>::fin` in logic context
  --> ghost_mapping.rs:13:21
   |
13 |     pearlite! { |_| ^x }
   |                     ^^

error: called prophetic logic function `<&mut T as creusot_contracts::logic::ops::Fin>::fin` in logic context
  --> ghost_mapping.rs:25:32
   |
25 |     pearlite! { forall<_x:Int> ^y == 1i32 }
   |                                ^^

error: aborting due to 2 previous errors

